analyzers/src/SonarAnalyzer.CSharp/Rules/NameOfShouldBeUsed.cs

```diff
@@ -33,53 +32,45 @@ namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     [Rule(DiagnosticId)]
-    public sealed class NameOfShouldBeUsed : NameOfShouldBeUsedBase<BaseMethodDeclarationSyntax>
+    public sealed class NameOfShouldBeUsed : NameOfShouldBeUsedBase<BaseMethodDeclarationSyntax, SyntaxKind, ThrowStatementSyntax>
     {
-       private static readonly HashSet<SyntaxKind> StringTokenTypes
-            = new HashSet<SyntaxKind>
+        private static readonly HashSet<SyntaxKind> StringTokenTypes = new HashSet<SyntaxKind>
             {
                 SyntaxKind.InterpolatedStringTextToken,
                 SyntaxKind.StringLiteralToken
             };
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
-
-        protected override DiagnosticDescriptor Rule { get; } = rule;
-
-        protected override bool IsCaseSensitive => true;
+        protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
 
         protected override bool IsStringLiteral(SyntaxToken t) => t.IsAnyKind(StringTokenTypes);
 
         protected override IEnumerable<string> GetParameterNames(BaseMethodDeclarationSyntax method)
         {
-            var paramGroups = method.ParameterList?.Parameters
-                .GroupBy(p => p.Identifier.ValueText);
-
-            if (paramGroups != null &&
-                paramGroups.Any(g => g.Count() != 1))
+            var paramGroups = method.ParameterList?.Parameters.GroupBy(p => p.Identifier.ValueText);
+            if (paramGroups != null && paramGroups.Any(g => g.Count() != 1))
```

Comment:
Something is not OK here, if paramGroups is null, we will have a null reference exception.

Reply:
Wow, you've spotted an old error. Thanks for the catch! But I am not sure that we could reproduce that, since parameter list is non-empty always (otherwise AST regards it as property)